tests/integration/plugins/test_poetry.py

```diff
@@ -242,7 +242,7 @@ def _get_system_python_interpreter(self) -> str | None:
     actions = lf.plan(Step.PRIME)
 
     out = pathlib.Path("out.txt")
-    with out.open(mode="w") as outfile, pytest.raises(errors.ScriptletRunError):
+    with out.open(mode="w") as outfile, pytest.raises(errors.PluginBuildError):
```

Comment:
Errors in the built-in handler executed using `craftctl default` now raise the same error as executing it directly.